Framework/PythonInterface/plugins/algorithms/BASISReduction.py

```diff
@@ -130,10 +131,32 @@ def PyInit(self):
         self.declareProperty('GroupDetectors', 'None',
                              StringListValidator(grouping_type),
                              'Switch for grouping detectors')
+        #
+        #  Normalization selector
+        #
+        title_flux_normalization = 'Flux Normalization'
+        self.declareProperty('DoFluxNormalization', True,
+                             direction=Direction.Input,
+                             doc='Do we normalize data by incoming flux?')
+        self.setPropertyGroup('DoFluxNormalization', title_flux_normalization)
+        if_flux_normalization = EnabledWhenProperty('DoFluxNormalization',
```

Comment:
Tempted to say avoid the PropertyGroup and make `none` a potential `flux_normalization_type`. That said, the selector you chose is more consistent with the rest of the BASISReduction dialog.

Reply:
I see what you mean, and I considered it. The disadvantage I saw with using only a selector is that the `none` option would be hidden, because the default option is `Monitor`. On the other hand, using both a selector _and_ a checkbox shows two options to the user at the same time. This is only relevant for newbie users. They may be a little baffled if not presented right away with an option for no flux normalization :smirk: